data-workflows/activity/install_activity_model.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Callable, List, Union
+import os
+
+from pynamodb.models import Model
+from pynamodb.attributes import UnicodeAttribute, NumberAttribute
+
+from utils.utils import get_current_timestamp
+
+LOGGER = logging.getLogger()
+
+
+class InstallActivityType(Enum):
+    DAY = 1
+    MONTH = 2
+    TOTAL = 3
+
+    def get_timestamp_format(self) -> Callable[[datetime], Union[int, None]]:
+        if self is InstallActivityType.TOTAL:
+            return lambda timestamp: None
+        return lambda timestamp: timestamp.replace(tzinfo=timezone.utc).timestamp() * 1000
+
+    def get_query_timestamp_projection(self) -> str:
+        return '1' if self is InstallActivityType.TOTAL else f'DATE_TRUNC(\'{self.name}\', timestamp)'
+
+    def get_type_timestamp_formatter(self) -> Callable[[datetime], str]:
+        if self is InstallActivityType.TOTAL:
+            return lambda timestamp: 'TOTAL:'
+        elif self is InstallActivityType.DAY:
+            return lambda timestamp: f'DAY:{timestamp.strftime("%Y%m%d")}'
+        return lambda timestamp: f'MONTH:{timestamp.strftime("%Y%m")}'
+
+    def get_timestamp_query_formatter(self) -> Callable[[datetime], datetime]:
+        return (lambda ts: ts) if self is InstallActivityType.DAY else (lambda ts: ts.replace(day=1))
+
+
+class InstallActivity(Model):
+    class Meta:
+        prefix = os.getenv('STACK_NAME')
+        region = os.getenv('AWS_REGION')
+        table_name = f'{prefix}-install-activity'
+
+    plugin_name = UnicodeAttribute(hash_key=True)
+    type_timestamp = UnicodeAttribute(range_key=True)
+    granularity = UnicodeAttribute(attr_name='type')
+    timestamp = NumberAttribute(null=True)
+    install_count = NumberAttribute()
+    last_updated_timestamp = NumberAttribute(default_for_new=get_current_timestamp)
+
+    def __eq__(self, other):
+        if isinstance(other, InstallActivity):
+            return (self.plugin_name, self.type_timestamp, self.granularity, self.timestamp, self.install_count) == \
+                   (other.plugin_name, other.type_timestamp, other.granularity, other.timestamp, other.install_count)
```

Comment:
```suggestion
            return ((self.plugin_name, self.type_timestamp, self.granularity, self.timestamp, self.install_count) == 
                    (other.plugin_name, other.type_timestamp, other.granularity, other.timestamp, other.install_count))
```

and format with e.g. `black`